src/SFA.DAS.AssessorService.Data/CertificateRepository.cs

```diff
@@ -399,45 +415,93 @@ public async Task<Certificate> UpdateProviderName(Guid id, string providerName)
             return certificate;
         }
 
-        private async Task UpdateCertificateLog(Certificate cert, string action, string username, string reasonForChange = null)
+        private async Task AddCertificateLog(Guid certificateId, string action, string status, DateTime eventTime, string certificateData, string username, int? batchNumber, string reasonForChange = null)
         {
             if (action != null)
             {
                 var certLog = new CertificateLog
                 {
-                    Action = action,
-                    CertificateId = cert.Id,
-                    EventTime = DateTime.UtcNow,
-                    Status = cert.Status,
                     Id = Guid.NewGuid(),
-                    CertificateData = cert.CertificateData,
+                    CertificateId = certificateId,
+                    Action = action,
+                    Status = status,
+                    EventTime = eventTime,
+                    CertificateData = certificateData,
                     Username = username,
-                    BatchNumber = cert.BatchNumber,
+                    BatchNumber = batchNumber,
                     ReasonForChange = reasonForChange
                 };
 
                 await _context.CertificateLogs.AddAsync(certLog);
             }
         }
 
-        public async Task UpdateStatuses(UpdateCertificatesBatchToIndicatePrintedRequest updateCertificatesBatchToIndicatePrintedRequest)
+        public async Task UpdatePrintStatus(Certificate certificate, int batchNumber, string printStatus, DateTime statusAt, bool changesCertificateStatus)
         {
-            var toBePrintedDate = DateTime.UtcNow;
+            var certificateBatchLog =
+                await _context.CertificateBatchLogs.FirstOrDefaultAsync(
+                    q => q.CertificateReference == certificate.CertificateReference && q.BatchNumber == batchNumber);
 
-            foreach (var certificateStatus in updateCertificatesBatchToIndicatePrintedRequest.CertificateStatuses)
+            if (certificateBatchLog != null)
             {
-                var certificate =
-                    await _context.Certificates.FirstAsync(q => q.CertificateReference == certificateStatus.CertificateReference);
+                if (changesCertificateStatus)
```

Comment:
should this check be done outside of the repository? I wouldn't expect to check whether to change the status inside a method specifically for updating the status

Reply:
I don't understand the question here - are you saying that the check should be done inside the repository method or outside? currently the check is done outside which to me seems correct.

Reply:
what I mean is, we shouldn't need to pass in the changesCertificateStatus flag and check it.
This method should only be called if ChangesCertificateStatus = true

Reply:
The method changes the print status; which is a combination of certificate status, batch log status and certificate logs; there is business logic for whether the certificate status is changed; simple logic for batch log status and the certificate logs are always updated, so I think it's correct that the flag is passed from the handler to the repository for updating the certificate status as the handler has the unit tests.

Reply:
ok, I think I see where you are coming from. Resolving this